cpp/src/arrow/type.h

```diff
@@ -1317,6 +1317,40 @@ class ARROW_EXPORT DayTimeIntervalType : public IntervalType {
   std::string name() const override { return "day_time_interval"; }
 };
 
+/// \brief Represents a number of days and milliseconds (fraction of day).
```

Comment:
Update docstring? (this looks like a paste of the `DayTimeIntervalType` docstring)